README.md

```diff
@@ -34,10 +35,108 @@ Installation
 ember install ember-sinon-qunit
 ```
 
-
 Usage
 ------------------------------------------------------------------------------
 
+Ember-Sinon-QUnit supports two different versions of the Ember-QUnit API:
+
+1. The new QUnit hooks API, which takes a `hooks` object then wires up sandbox 
+creation and restoration to `beforeEach` and `afterEach` of the module, respectively.
+1. The classic API, which automatically wires up sandbox creation and restoration 
+to `QUnit.testStart` and `QUnit.testDone` respectively.
+
+### QUnit `hooks` API
+
+To use, import the setup method from within your test file and execute it.
+
+```js
+import { setupSinon } from 'ember-sinon-qunit/test-support';
+
+...
+
+module('my module', function(hooks) {
+  setupSinon(hooks);
+
+  test('my test', function(assert) {
+    ...
+  })
+})
+```
+
+This will automatically wire-up the sandbox setup & restoration to the module's `beforeEach` and `afterEach` respectively.
+
+### Classic API
+
+To use, import the setup method from within your `tests/test-helper.js` file and execute it.
+
+```js
+import setupSinon from 'ember-sinon-qunit/test-support/setup-global-sinon';
+
+...
+
+setupSinon();
+```
+
+This will automatically wire-up the sandbox setup & restoration to QUnit `testStart` and `testDone` respectively.
+
+#### Accessing Sinon from Within Tests
+
+In each test you will be able to access the same sandboxed version of sinon via the `sinon` object available as an import in your tests:
+
+```js
+import sinon from 'sinon';
+
+...
+
+test('very important test happening here', function(assert) {
+  const spy = sinon.spy();
+
+  ...
+});
+```
+
+The `sinon` object is automatically sandboxed to each specific test, allowing you to 
+safely create mocks for your tests without worrying about any overrides leaking between each test.
+
+Migrating To Ember-Sinon-QUnit
+------------------------------------------------------------------------------
+
+The above functionality replaces previous features within Ember-Sinon-QUnit, 
+as well as the sister addons [`ember-sinon-sinoff`](https://github.com/scalvert/ember-sinon-sinoff) 
+and [`ember-sinon-sandbox`](https://github.com/scalvert/ember-sinon-sandbox). 
+Below, you will find simple instructions for migrating from each of these feature sets to the new patterns.
+
+### Migration from Sinon 5+
+
+1. Import and consume `setupSinon` for the [Hooks API](#qunit-hooks-api) or [Classic API](#classic-api).
+1. Remove any manual calls to `sinon.restore()`. It won't hurt to leave them, but they are redundant now!
```

Comment:
Any tests that are not doing `import sinon from 'sinon'` will also have to do so.

Reply:
Wouldn't they already be doing that if they're using Sinon?

Reply:
No, because for older versions of `sinon` it was exposed as a global on `self`.  So depending on the version you were on you may have to add the import.